TestContainers 'org.testcontainers.containers.ContainerLaunchException' on arm64 [HZ-3056]

_master_ (commit 61aab65ede79b58bcf416ddfdb6c48f523064d0b)
Failed on OpenJDK11: https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK11-arm64-nightly/25/

Tests which failed:

[com.hazelcast.mapstore.mssql.MSSQLGenericMapStoreTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK11-arm64-nightly/lastSuccessfulBuild/testReport/junit/com.hazelcast.mapstore.mssql/MSSQLGenericMapStoreTest/___/)
[com.hazelcast.jet.sql.impl.connector.jdbc.mssql.MSSQLAllTypesInsertJdbcSqlConnectorTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK11-arm64-nightly/lastSuccessfulBuild/testReport/junit/com.hazelcast.jet.sql.impl.connector.jdbc.mssql/MSSQLAllTypesInsertJdbcSqlConnectorTest/___/)
[com.hazelcast.jet.sql.impl.connector.jdbc.mssql.MSSQLAllTypesSelectJdbcSqlConnectorTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK11-arm64-nightly/lastSuccessfulBuild/testReport/junit/com.hazelcast.jet.sql.impl.connector.jdbc.mssql/MSSQLAllTypesSelectJdbcSqlConnectorTest/___/)
[com.hazelcast.jet.sql.impl.connector.jdbc.mssql.MSSQLCreateDataConnectionJdbcSqlConnectorTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK11-arm64-nightly/lastSuccessfulBuild/testReport/junit/com.hazelcast.jet.sql.impl.connector.jdbc.mssql/MSSQLCreateDataConnectionJdbcSqlConnectorTest/___/)
[com.hazelcast.jet.sql.impl.connector.jdbc.mssql.MSSQLDeleteJdbcSqlConnectorTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK11-arm64-nightly/lastSuccessfulBuild/testReport/junit/com.hazelcast.jet.sql.impl.connector.jdbc.mssql/MSSQLDeleteJdbcSqlConnectorTest/___/)
[com.hazelcast.jet.sql.impl.connector.jdbc.mssql.MSSQLDeletePredicatePushDownJdbcSqlConnectorTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK11-arm64-nightly/lastSuccessfulBuild/testReport/junit/com.hazelcast.jet.sql.impl.connector.jdbc.mssql/MSSQLDeletePredicatePushDownJdbcSqlConnectorTest/___/)
[com.hazelcast.jet.sql.impl.connector.jdbc.mssql.MSSQLInsertJdbcSqlConnectorTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK11-arm64-nightly/lastSuccessfulBuild/testReport/junit/com.hazelcast.jet.sql.impl.connector.jdbc.mssql/MSSQLInsertJdbcSqlConnectorTest/___/)
[com.hazelcast.jet.sql.impl.connector.jdbc.mssql.MSSQLJdbcJoinTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK11-arm64-nightly/lastSuccessfulBuild/testReport/junit/com.hazelcast.jet.sql.impl.connector.jdbc.mssql/MSSQLJdbcJoinTest/___/)
[com.hazelcast.jet.sql.impl.connector.jdbc.mssql.MSSQLJdbcSqlConnectorStabilityTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK11-arm64-nightly/lastSuccessfulBuild/testReport/junit/com.hazelcast.jet.sql.impl.connector.jdbc.mssql/MSSQLJdbcSqlConnectorStabilityTest/___/)
[com.hazelcast.jet.sql.impl.connector.jdbc.mssql.MSSQLMappingJdbcSqlConnectorTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK11-arm64-nightly/lastSuccessfulBuild/testReport/junit/com.hazelcast.jet.sql.impl.connector.jdbc.mssql/MSSQLMappingJdbcSqlConnectorTest/___/)
[com.hazelcast.jet.sql.impl.connector.jdbc.mssql.MSSQLPredicatePushDownJdbcSqlConnectorTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK11-arm64-nightly/lastSuccessfulBuild/testReport/junit/com.hazelcast.jet.sql.impl.connector.jdbc.mssql/MSSQLPredicatePushDownJdbcSqlConnectorTest/___/)
[com.hazelcast.jet.sql.impl.connector.jdbc.mssql.MSSQLSelectJdbcSqlConnectorTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK11-arm64-nightly/lastSuccessfulBuild/testReport/junit/com.hazelcast.jet.sql.impl.connector.jdbc.mssql/MSSQLSelectJdbcSqlConnectorTest/___/)
[com.hazelcast.jet.sql.impl.connector.jdbc.mssql.MSSQLUpdateJdbcSqlConnectorTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK11-arm64-nightly/lastSuccessfulBuild/testReport/junit/com.hazelcast.jet.sql.impl.connector.jdbc.mssql/MSSQLUpdateJdbcSqlConnectorTest/___/)
[com.hazelcast.jet.sql.impl.connector.jdbc.mssql.MSSQLUpdatePredicatePushDownJdbcSqlConnectorTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK11-arm64-nightly/lastSuccessfulBuild/testReport/junit/com.hazelcast.jet.sql.impl.connector.jdbc.mssql/MSSQLUpdatePredicatePushDownJdbcSqlConnectorTest/___/)

<details><summary>Example Stacktrace:</summary>

```
org.testcontainers.containers.ContainerLaunchException: Container startup failed for image mcr.microsoft.com/mssql/server:2017-CU12
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:349)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:322)
	at com.hazelcast.test.jdbc.MSSQLDatabaseProvider.createDatabase(MSSQLDatabaseProvider.java:39)
	at com.hazelcast.jet.sql.impl.connector.jdbc.JdbcSqlTestSupport.initialize(JdbcSqlTestSupport.java:70)
	at com.hazelcast.jet.sql.impl.connector.jdbc.JdbcSqlTestSupport.initialize(JdbcSqlTestSupport.java:65)
	at com.hazelcast.mapstore.mssql.MSSQLGenericMapStoreTest.beforeClass(MSSQLGenericMapStoreTest.java:31)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at com.hazelcast.test.AfterClassesStatement.evaluate(AfterClassesStatement.java:41)
	at com.hazelcast.test.OverridePropertyRule$1.evaluate(OverridePropertyRule.java:66)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:334)
	... 21 more
Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:553)
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:344)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
	... 22 more
Caused by: org.testcontainers.shaded.org.awaitility.core.ConditionTimeoutException: Lambda expression in org.testcontainers.containers.GenericContainer: expected the predicate to return <true> but it returned <false> for input of <InspectContainerResponse(args=[], config=ContainerConfig(attachStderr=false, attachStdin=false, attachStdout=false, cmd=[/opt/mssql/bin/sqlservr], domainName=, entrypoint=null, env=[ACCEPT_EULA=Y, SA_PASSWORD=A_Str0ng_Required_Password, PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin], exposedPorts=[1433/tcp], hostName=c8cdfa3eb8cb, image=mcr.microsoft.com/mssql/server:2017-CU12, labels={com.microsoft.product=Microsoft SQL Server, com.microsoft.version=14.0.3045.24, org.testcontainers=true, org.testcontainers.lang=java, org.testcontainers.sessionId=9ec9eb88-ec74-4bfa-a473-f4863b62a170, org.testcontainers.version=1.18.3, vendor=Microsoft}, macAddress=null, networkDisabled=null, onBuild=null, stdinOpen=false, portSpecs=null, stdInOnce=false, tty=false, user=, volumes=null, workingDir=, healthCheck=null), created=2023-08-24T05:11:44.276458555Z, driver=overlay2, execDriver=null, hostConfig=HostConfig(binds=[], blkioWeight=0, blkioWeightDevice=null, blkioDeviceReadBps=null, blkioDeviceWriteBps=null, blkioDeviceReadIOps=null, blkioDeviceWriteIOps=null, memorySwappiness=null, nanoCPUs=0, capAdd=null, capDrop=null, containerIDFile=, cpuPeriod=0, cpuRealtimePeriod=0, cpuRealtimeRuntime=0, cpuShares=0, cpuQuota=0, cpusetCpus=, cpusetMems=, devices=null, deviceCgroupRules=null, deviceRequests=null, diskQuota=null, dns=null, dnsOptions=null, dnsSearch=null, extraHosts=[], groupAdd=null, ipcMode=shareable, cgroup=, links=[], logConfig=LogConfig(type=json-file, config={}), lxcConf=null, memory=0, memorySwap=0, memoryReservation=0, kernelMemory=0, networkMode=default, oomKillDisable=false, init=null, autoRemove=false, oomScoreAdj=0, portBindings={1433/tcp=[Lcom.github.dockerjava.api.model.Ports$Binding;@697b8f99}, privileged=false, publishAllPorts=false, readonlyRootfs=false, restartPolicy=no, ulimits=[Ulimit(name=nofile, soft=32768, hard=65536)], cpuCount=0, cpuPercent=0, ioMaximumIOps=0, ioMaximumBandwidth=0, volumesFrom=[], mounts=null, pidMode=, isolation=null, securityOpts=null, storageOpt=null, cgroupParent=, volumeDriver=, shmSize=67108864, pidsLimit=null, runtime=runc, tmpFs=null, utSMode=, usernsMode=, sysctls=null, consoleSize=[0, 0], cgroupnsMode=host), hostnamePath=/var/lib/docker/containers/c8cdfa3eb8cb4f70cd47f3d405a0716e20551f467f0ffca8e7ce855e7b6dd8d1/hostname, hostsPath=/var/lib/docker/containers/c8cdfa3eb8cb4f70cd47f3d405a0716e20551f467f0ffca8e7ce855e7b6dd8d1/hosts, logPath=/var/lib/docker/containers/c8cdfa3eb8cb4f70cd47f3d405a0716e20551f467f0ffca8e7ce855e7b6dd8d1/c8cdfa3eb8cb4f70cd47f3d405a0716e20551f467f0ffca8e7ce855e7b6dd8d1-json.log, id=c8cdfa3eb8cb4f70cd47f3d405a0716e20551f467f0ffca8e7ce855e7b6dd8d1, sizeRootFs=null, imageId=sha256:4095d6d460cdcf85ec19b5fd15d354e2827419cc416f311c54cc9671a90a2b1d, mountLabel=, name=/jovial_napier, restartCount=0, networkSettings=NetworkSettings(bridge=, sandboxId=c1c563fa873cce5eefd26bcfd2b6062d8712bae1fa4db3853a066207d72e45b9, hairpinMode=false, linkLocalIPv6Address=, linkLocalIPv6PrefixLen=0, ports={}, sandboxKey=/var/run/docker/netns/c1c563fa873c, secondaryIPAddresses=null, secondaryIPv6Addresses=null, endpointID=, gateway=, portMapping=null, globalIPv6Address=, globalIPv6PrefixLen=0, ipAddress=, ipPrefixLen=0, ipV6Gateway=, macAddress=, networks={bridge=ContainerNetwork(ipamConfig=null, links=[], aliases=null, networkID=78a7a635f89ff4ec6027ca3cf978e914fab00b0b2ba812fb811728e8aa4ee840, endpointId=, gateway=, ipAddress=, ipPrefixLen=0, ipV6Gateway=, globalIPv6Address=, globalIPv6PrefixLen=0, macAddress=)}), path=/opt/mssql/bin/sqlservr, processLabel=, resolvConfPath=/var/lib/docker/containers/c8cdfa3eb8cb4f70cd47f3d405a0716e20551f467f0ffca8e7ce855e7b6dd8d1/resolv.conf, execIds=null, state=InspectContainerResponse.ContainerState(status=exited, running=false, paused=false, restarting=false, oomKilled=false, dead=false, pid=0, exitCode=1, error=, startedAt=2023-08-24T05:11:44.787949213Z, finishedAt=2023-08-24T05:11:44.787750618Z, health=null), volumes=null, volumesRW=null, node=null, mounts=[], graphDriver=GraphDriver(name=overlay2, data=GraphData(rootDir=null, deviceId=null, deviceName=null, deviceSize=null, dir=null)), platform=linux)> within 5 seconds.
	at org.testcontainers.shaded.org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:167)
	at org.testcontainers.shaded.org.awaitility.core.AbstractHamcrestCondition.await(AbstractHamcrestCondition.java:86)
	at org.testcontainers.shaded.org.awaitility.core.ConditionFactory.until(ConditionFactory.java:985)
	at org.testcontainers.shaded.org.awaitility.core.ConditionFactory.until(ConditionFactory.java:691)
	at org.testcontainers.shaded.org.awaitility.core.ConditionFactory.until(ConditionFactory.java:708)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:455)
	... 24 more

```
</details>

--- a/hazelcast/src/test/java/com/hazelcast/test/jdbc/MSSQLDatabaseProvider.java
+++ b/hazelcast/src/test/java/com/hazelcast/test/jdbc/MSSQLDatabaseProvider.java
@@ -18,6 +18,8 @@ package com.hazelcast.test.jdbc;
 
 import org.testcontainers.containers.MSSQLServerContainer;
 
+import static com.hazelcast.test.HazelcastTestSupport.assumeNoArm64Architecture;
+
 public class MSSQLDatabaseProvider implements TestDatabaseProvider {
 
     public static final String TEST_MSSQLSERVER_VERSION = System.getProperty("test.mssqlserver.version", "2017-CU12");
@@ -28,6 +30,7 @@ public class MSSQLDatabaseProvider implements TestDatabaseProvider {
 
     @Override
     public String createDatabase(String dbName) {
+        assumeNoArm64Architecture();
         container = new MSSQLServerContainer<>("mcr.microsoft.com/mssql/server:" + TEST_MSSQLSERVER_VERSION);
         container.acceptLicense()
                  // See https://learn.microsoft.com/en-us/sql/connect/jdbc/using-basic-data-types?view=sql-server-ver16

